src/components/shared/register/RegisterForm.tsx

```diff
@@ -1,92 +1,144 @@
 import { Button, Error, Input } from 'components/lib';
-import { FC, useCallback, useMemo, useState } from 'react';
-import { useForm } from 'react-hook-form';
+import { FC, useCallback, useState } from 'react';
+import { FieldErrors, useForm } from 'react-hook-form';
 import { REGISTER_FORM_FIELDS } from './RegisterForm.constants';
 import { TSubmitFormEvent } from 'types';
 import { PasswordValidityLevel, UsernameValidityError } from '..';
 
-const { name, bio, login, password } = REGISTER_FORM_FIELDS;
+const { name, bio, login, password, email } = REGISTER_FORM_FIELDS;
 
-export const RegisterForm: FC = () => {
-  const [currentStep, setCurrentStep] = useState(0);
+export const RegisterForm: FC<{ resetFormType: () => void; formType: 'login' | 'register' }> = ({
+  resetFormType,
+  formType,
```

Comment:
you do NOT need this formType prop. Form type will always be register when RegisterForm is rendered